Document changes for SHOW INDEXES/CONSTRAINTS columns (#1523)
neo4j/neo4j-documentation: cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc

Constraints used for string properties require an additional text index to cover the string queries properly.
Constraints used for point properties require an additional point index to cover the spatial queries properly.

a|
label:functionality[]
label:removed[]
[source, cypher, role="noheader"]
----
SHOW INDEXES YIELD uniqueness
----
a|

> The `uniqueness` output has been removed altogether with the concept of index uniqueness, as it actually belongs to the constraint and not the index.

Reviewer: ```suggestion
The `uniqueness` output has been removed along with the concept of index uniqueness, as it actually belongs to the constraint and not the index.
```

In my opinion `altogether with ` should be replaced by either `along with` or `altogether, along with`

Reviewer 2: I agree, thanks for the suggestion Phil

Constraints used for string properties require an additional text index to cover the string queries properly.
Constraints used for point properties require an additional point index to cover the spatial queries properly.

a|
label:functionality[]
label:removed[]
[source, cypher, role="noheader"]
----
SHOW INDEXES YIELD uniqueness
----
a|

> The `uniqueness` output has been removed, as the concept of index uniqueness are removed.

Reviewer: ```suggestion
The `uniqueness` output has been removed, as the concept of index uniqueness has been removed.
```

> The `uniqueness` output has been removed, as the concept of index uniqueness has been removed.

Reviewer 2: ```suggestion
The `uniqueness` output has been removed altogether with the concept of index uniqueness, as it actually belongs to the constraint and not the index.
```

> The uniqueness concept belongs with the constraint and not the index.

Reviewer 2: ```suggestion

```

> The `uniqueness` output has been removed, as the concept of index uniqueness has been removed.
> The uniqueness concept belongs with the constraint and not the index.

Author: ```suggestion
The `uniqueness` output has been removed altogether with the concept of index uniqueness, as it actually belongs to the constraint and not the index.
```

> A new column, `owningConstraint`, is introduced to indicate whether an index belongs to a constraint or not.

Reviewer 2: ```suggestion
The new column `owningConstraint` was introduced to indicate whether an index belongs to a constraint or not.
```

> A new column, `owningConstraint`, is introduced keep information about the index belonging to a constraint or not.

Reviewer: ```suggestion
A new column, `owningConstraint`, is introduced to indicate whether an index belongs to a constraint or not.
```

a|
label:functionality[]
label:removed[]
[source, cypher, role="noheader"]
----
SHOW CONSTRAINTS YIELD ownedIndexId
----
a|

> The `ownedIndexId` output has been removed. Replaced by new `ownedIndex` column.

Reviewer 2: ```suggestion
The `ownedIndexId` output has been removed and replaced by the new `ownedIndex` column.
```

a|
The default index type is changed from B-tree to range index.

a|
label:functionality[]
label:updated[]
[source, cypher, role="noheader"]
----
SHOW INDEXES
----
a|
The new column `owningConstraint` was added and will be returned by default from now on.
It will list the name of the constraint that the index is associated with or `null`, in case it is not associated with any constraint.

a|
label:functionality[]
label:updated[]
[source, cypher, role="noheader"]
----
SHOW CONSTRAINTS
----
a|

> The new column `ownedIndex` was added and will be returned by default from now on.

Reviewer 3: Why this inconsistency?

```
ownedIndex
owningConstraint
```

`owned` vs `owning` ???

Author: I believe this has been answered on slack :)

a|
The default index type is changed from B-tree to range index.

a|
label:functionality[]
label:updated[]
[source, cypher, role="noheader"]
----
SHOW INDEXES
----
a|

> A new column is added, `owningConstraint`. The column is returned by default.

Reviewer 2: ```suggestion
The new column `owningConstraint` was added and will be returned by default from now on.
```

> This column returns the name of the constraint the index is associated to, or `null` if it is not associated with any constraint.

Reviewer: ```suggestion
This column returns the name of the constraint the index is associated with, or `null` if it is not associated with any constraint.
```

> This column returns the name of the constraint the index is associated with, or `null` if it is not associated with any constraint.

Reviewer 2: ```suggestion
It will list the name of the constraint that the index is associated with or `null`, in case it is not associated with any.
```

Author: I feel like the sentence ends very abruptly on the `any`, but that might just be me?
 ```suggestion
It will list the name of the constraint that the index is associated with or `null`, in case it is not associated with any constraint.
```

Reviewer 2: I think in terms of grammar it's fine to end with "any", but we can go for your suggestion with no problems! It's also correct and we can avoid any other readers finding it weird. Thanks!

a|
label:functionality[]
label:updated[]
[source, cypher, role="noheader"]
----
SHOW CONSTRAINTS
----
a|

> A new column is added, `ownedIndex`. The column is returned by default.

Reviewer 2: ```suggestion
The new column `ownedIndex` was added and will be returned by default from now on.
```

> This column returns the name of the index associated to the constraint, or `null` if no index is associated with the constraint.

Reviewer: ```suggestion
This column returns the name of the index associated with the constraint, or `null` if no index is associated with the constraint.
```

> This column returns the name of the index associated with the constraint, or `null` if no index is associated with the constraint.

Reviewer 2: ```suggestion
It will list the name of the index associated with a constraint or `null`, in case no index is associated with it.
```

Author: ```suggestion
It will list the name of the index associated with the constraint or `null`, in case no index is associated with it.
```
you kept the `the` for the index equivalent sentence (which I think sounds better)

Reviewer 2: Agreed!
